controllers/componentversion_controller.go

```diff
@@ -82,14 +83,73 @@ func (r *ComponentVersionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}, fmt.Errorf("attempted to verify component, but the digest didn't match")
 	}
 
+	// reconcile the version before calling reconcile
+	update, err := r.checkVersion(ctx, component)
+	if err != nil {
+		return ctrl.Result{
+			RequeueAfter: component.GetRequeueAfter(),
+		}, fmt.Errorf("failed to check version: %w", err)
+	}
+
+	if !update {
+		return ctrl.Result{}, nil
+	}
+
 	return r.reconcile(ctx, component)
 }
 
+func (r *ComponentVersionReconciler) checkVersion(ctx context.Context, obj *v1alpha1.ComponentVersion) (bool, error) {
+	log := log.FromContext(ctx).WithName("ocm-component-version-reconcile")
+
+	// get given semver constraint
+	constraint, err := semver.NewConstraint(obj.Spec.Version.Semver)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse given semver constraint: %w", err)
+	}
+
+	// get current reconciled version
+	reconciledVersion := "0.0.0"
+	if obj.Status.ReconciledVersion != "" {
+		reconciledVersion = obj.Status.ReconciledVersion
+	}
+	current, err := semver.NewVersion(reconciledVersion)
```

Comment:
@nikimanoledaki I don't see where you set this as a new version to be reconciled? I think you have to patch the object with this new version in order for it to pick it up. Right now, you just say, update is required, but you don't update the state of the object so it reflects what version to reconcile. It will be the same version as before.

Reply:
Sorry for the delay in getting back to you, finally picked up this work again. The patch is there now 😊

Reply:
I'll take a look today. :)